radixdlt/src/main/java/com/radixdlt/consensus/Vertex.java

```diff
@@ -57,12 +57,40 @@
 		this.atom = null;
 	}
 
-	public Vertex(QuorumCertificate qc, Round round, Atom atom) {
+	private Vertex(QuorumCertificate qc, Round round, Atom atom) {
 		this.qc = qc;
-		this.round = Objects.requireNonNull(round);
+		this.round = round;
 		this.atom = atom;
 	}
 
+	public static Vertex createGenesis(Atom atom) {
+		return new Vertex(null, Round.of(0), atom);
+	}
+
+	public static Vertex createVertex(QuorumCertificate qc, Round round, Atom atom) {
+		Objects.requireNonNull(qc);
+		Objects.requireNonNull(round);
+
+		if (round.number() == 0) {
+			throw new IllegalArgumentException("Only genesis can have round 0.");
+		}
+
+		return new Vertex(qc, round, atom);
+	}
+
+	// TODO: Replace with a better type over EUID wrapping hashCode()
+	public EUID getId() {
+		return new EUID(this.hashCode());
```

Comment:
Can we fix this?  Using a tiny 32 bit hash of dubious merit seems almost certain to bite us hard at some point in the near future.

Given that this class is now serializable, it should not be too hard to calculate a proper hash / EUID.

Reply:
Yes, we also perhaps need a discussion on whether 128 bits is enough vs 256 bits like bitcoin